install-poetry.py

```diff
@@ -277,6 +262,64 @@ def temporary_directory(*args, **kwargs):
 POST_MESSAGE_CONFIGURE_WINDOWS = """"""
 
 
+class PoetryInstallationError(RuntimeError):
+    def __init__(self, return_code: int = 0, log: Optional[str] = None):
+        super(PoetryInstallationError, self).__init__()
+        self.return_code = return_code
+        self.log = log
+
+
+class VirtualEnvironment:
+    def __init__(self, path: Path) -> None:
+        self._path = path
+        self._python = self._path.joinpath(
+            "Scripts/python.exe" if WINDOWS else "bin/python"
+        )
+
+    @property
+    def path(self):
+        return self._path
+
+    @classmethod
+    def make(cls, target: Path) -> "VirtualEnvironment":
+        import venv
+
+        builder = venv.EnvBuilder(clear=True, with_pip=True, symlinks=False)
+        builder.ensure_directories(target)
+        builder.create(target)
+
+        env = cls(target)
+
+        # we do this here to ensure that outdated system default pip does not trigger older bugs
+        env.pip("install", "--upgrade", "pip")
+
+        return cls(target)
```

Comment:
Is there a reason to not return `env`? Seems like the class is just a simple holder for the `target`/`path` so there's no reason not to re-use it.